### PR TITLE
Use cargo tomlfmt

### LINKED ISSUE #634: Use tomlfmt
https://github.com/tbrand/cargo-tomlfmt to fmt Cargo.toml files

Blocked by tbrand/cargo-tomlfmt#12.

### PR DESCRIPTION
**Pull request by luckysori**
Two `cargo make` tasks have been added to the `Makefile.toml`:
1. `cargo make cargo-toml-format`, which checks the format of all `Cargo.toml` files in the repository and modifies them when necessary.
2. `cargo make check-cargo-toml-format`, which checks the format of all `Cargo.toml` files in the repository
and fails as it finds one which is wrongly formatted.

This last one is used in the pre-commit hook, so that only correctly formatted `Cargo.toml` files can be commited.

Fixes #634.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,19 +1,5 @@
...
-    "vendor/bitcoin_support",
-    "vendor/bitcoin_witness",
-    "vendor/ethereum_support",
-    "vendor/key_gen",
-    "vendor/secp256k1_support",
-    "vendor/tc_bitcoincore_client",
-    "vendor/tc_web3_client",
-    "vendor/noise_handshake"
-]
+members = ["application/comit_node", "application/ledger_query_service", "vendor/bam", "vendor/bitcoin_rpc_test_helpers", "vendor/bitcoin_support", "vendor/bitcoin_witness", "vendor/ethereum_support", "vendor/key_gen", "vendor/secp256k1_support", "vendor/tc_bitcoincore_client", "vendor/tc_web3_client", "vendor/noise_handshake"]
```
Comment (thomaseizinger) [inline Cargo.toml]: :x: This is actually a no-go for me.

Can we open an issue that this is formatted differently?

Comment (D4nte) [reply Cargo.toml]: Ah yeah, I remember this. I think I tweeted with the creator about it. Please open an issue and see if we can configure it :)
OR, we don't apply the formmating on the meta cargo.toml

Comment (D4nte) [reply Cargo.toml]: Ah yeah, I remember this. I think I tweeted with the creator about it. Please open an issue and see if we can configure it :)
OR, we don't apply the formmating on the meta cargo.toml

File: Makefile.toml [hunk 2]
```diff
@@ -129,6 +129,26 @@ find . -not -path '*/node_modules/*' -name '*.js' | xargs ./api_tests/node_modul
 '''
 ]
 
+[tasks.cargo-toml-format]
+description = "Runs cargo tomlfmt to format Cargo.toml files."
+workspace = false
+install_crate = { crate_name = "cargo-tomlfmt", binary = "cargo-tomlfmt", test_arg = "--help" }
```
Comment (D4nte) [inline Makefile.toml]: this may not be needed

Comment (luckysori) [reply Makefile.toml]: It's just installing `cargo-tomlfmt` if it can't find it, right? Do we not want this to happen?

**Comment by luckysori** [comment]
Blocked until tbrans/cargo-tomlfmt#12 is resolved.

File: Makefile.toml (hunk 2 above)
Comment (thomaseizinger) [reply Makefile.toml]: I think the attribute has defaults but I am not sure they apply here.

Comment (D4nte) [reply Makefile.toml]: as explained IRL. cargo-make may be clever enough not to need this line

**Comment by D4nte** [comment]
a `format` tasks that does all the format would be good. Practical especially if you do a big change (add dependencies, update rust and js).

**Comment by bonomat** [comment]
I close this PR for now as it's been stale for a long time, has many merge conflicts and [tomlfmt](https://github.com/tbrand/cargo-tomlfmt) does not seem to have any active development currently. I would say we just leave the ticket #634 itself blocked in icebox and revisit it in the future.

**Pull request closed without merging**